util/util.go

```diff
@@ -493,15 +494,30 @@ func ClusterToObjectsMapper(c client.Client, ro runtime.Object, scheme *runtime.
 		return nil, err
```

Comment:
On line 486, where is this called? Shouldn't the object be a client.Object so we can remove the check for metadata/namespace?

Reply:
Usage looks like this:
```golang
	clusterToMachines, err := util.ClusterToObjectsMapper(mgr.GetClient(), &clusterv1.MachineList{}, mgr.GetScheme())
	clusterToMachineDeployments, err := util.ClusterToObjectsMapper(mgr.GetClient(), &clusterv1.MachineDeploymentList{}, mgr.GetScheme())
	clusterToMachineSets, err := util.ClusterToObjectsMapper(mgr.GetClient(), &clusterv1.MachineSetList{}, mgr.GetScheme())
	clusterToMachinePools, err := util.ClusterToObjectsMapper(mgr.GetClient(), &expv1.MachinePoolList{}, mgr.GetScheme())
	clusterToDockerMachines, err := util.ClusterToObjectsMapper(mgr.GetClient(), &infrav1.DockerMachineList{}, mgr.GetScheme())
	clusterToDockerMachinePools, err := util.ClusterToObjectsMapper(mgr.GetClient(), &infrav1exp.DockerMachinePoolList{}, mgr.GetScheme())
```

I could only change it to `client.ObjectList`.

How would have changing it to client.Object made the namespace check redundant? I think GetNamespace could return an empty namespace for a namespaced and a cluster-wide object.

Reply:
clientList sounds good, that should avoid the casting to ListInterface above

Reply:
Ah that typecast. Thought you're referring to the `isAPINamespaced` check

Reply:
Fine for me to change it, but requires a breaking change and thus a new func + deprecation (post v1.0)

Reply:
Let's do it now?

Reply:
I'll open a separate PR to change the signature. The rest of the PR is non-breaking but requires changes to the fakeclient which will take a bit.

Reply:
> https://github.com/kubernetes-sigs/cluster-api/pull/5391